R/faasr_client_tools.R

```diff
@@ -539,3 +539,294 @@ faasr_unset_workflow_timer <- function(target=NULL,...){
 }
 .faasr_user$operations$unset_workflow_timer <- faasr_unset_workflow_timer
 
+faasr_get_log_df <- function(){
+  # get the "svc" by using "faasr_get_svc" and define the required variables.
+  svc <- .faasr_get_svc()
+  faasr_wd <- svc$wd
+  if (!dir.exists(faasr_wd)){
+    faasr_wd <- getwd()
+  }
+  on.exit(setwd(faasr_wd))
+  faasr <- svc$json
+  cred <- faasr_collect_sys_env(faasr,svc$cred)
+  
+  setwd(faasr_wd)
+
+
+  storage_info <- faasr_get_storage_instance(faasr, cred)
+  log_server <- storage_info$server
+  s3 <- storage_info$s3
+
+
+  # Initialize lists to store data
+  uuids <- c()
+  dates <- c()
+  function_names <- c()
+  log_file_depth <- length(unlist(strsplit(faasr$FaaSrLog, "/")))
+  
+  # Initialize the continuation token
+  continuation_token <- NULL
+  repeat {
+    # Get the list of objects, using continuation token if not the first request
+    list_args <- list(Bucket = log_server$Bucket, Prefix = faasr$FaaSrLog)
+    if (!is.null(continuation_token)) {
+      list_args$ContinuationToken <- continuation_token
+    }
+    check_log_file <- do.call(s3$list_objects_v2, list_args)
+
+    # Process the files
+    if (check_log_file$KeyCount > 0){
+      for(content in check_log_file$Contents) {
+        key <- content$Key
+        key_parts <- strsplit(key, "/")[[1]]
+        if(length(key_parts) > (log_file_depth + 1)) {  
+          uuid <- key_parts[log_file_depth + 1]
+          file_name <- key_parts[length(key_parts)]
+          if(grepl("\\.txt$", file_name)) {  
+            uuids <- c(uuids, uuid)
+            dates <- c(dates, content$LastModified)
+            function_names <- c(function_names, file_name)
+          }
+        }
+      }
+    }
+    # Check if there is more data to fetch
+    if (!is.null(check_log_file$NextContinuationToken) && length(check_log_file$NextContinuationToken) > 0) {
+
+      continuation_token <- check_log_file$NextContinuationToken
+    } else {
+      break
+    }
+  }
+
+  # Format dates and create data frame if any data exists
+  if (length(uuids) > 0) {
+    dates <- as.POSIXct(dates, format="%Y-%m-%d %H:%M:%S", tz="")
```

Comment:
`dates <- as.POSIXct(dates, format="%Y-%m-%d %H:%M:%S", tz="")`
This code makes following error:

`Error in as.POSIXct.numeric(dates, format = "%Y-%m-%d %H:%M:%S", tz = "") : 
  'origin' must be supplied`